File: hc-mdm-web/src/main/webapp/resources/app/view/bascolor/BasColorController.js
/**
 * Description: 颜色信息表(来源于集团主数据 pro_color )Controller
 * All rights Reserved, Designed ByHc
 * Copyright:   Copyright(C) 2014-2015
 * Company:     Wonhigh.
 * @author:     panzh
 * @date:  2015-03-13 09:39:53
 * @version 1.0.0
 */
Ext.define('Hc_mdm.view.bascolor.BasColorController', {
    extend: 'Hc_Common.view.BaseSimplePageController',
    alias: 'controller.bascolor',
    init: function() {
        this.callParent(arguments);
        this.getReferences().mastergrid.on('edit', this.gridEdit, this);
    },
    gridEdit: function(obj, e) {
        if (e.field == 'colorSeriesName') {
            var value = Ext.getCmp('csCode').getRawValue(); //获取下拉框当前显示的值
            lengths = value.length - e.value.length - 2;
            value = Ext.util.Format.substr(value, 0, lengths);
            e.record.set('colorSeriesCode', value); //自动填充色系名称
        }
    }

});

File: hc-mdm-web/src/main/java/com/hc/scm/mdm/web/controller/BasColorController.java
package com.hc.scm.mdm.web.controller;


import com.hc.scm.common.base.service.BaseCrudService;
import com.hc.scm.common.base.web.BaseCrudController;
import com.hc.scm.mdm.dao.entity.BasColor;
import com.hc.scm.mdm.service.BasColorService;

import javax.annotation.Resource;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;

/**
 * Description: 请写出类的用途
 * All rights Reserved, Designed Byhc* Copyright:   Copyright(C) 2014-2015
 * Company:     Wonhigh.
 * @author:     panzh
 * @date:  2015-03-13 09:39:53
 * @version 1.0.0
 */
@Controller
@RequestMapping("/bas_color")
public class BasColorController extends BaseCrudController<BasColor> {
    @Resource
    private BasColorService basColorService;

    @Override
    public BaseCrudService init() {
        return basColorService;
    }
}